multiapps-controller-web/src/main/java/org/cloudfoundry/multiapps/controller/web/api/impl/FilesApiServiceImpl.java

```diff
@@ -181,21 +193,59 @@ private FileEntry uploadFileFromUrl(String spaceGuid, String namespace, String f
         }
     }
 
-    protected HttpClient buildHttpClient() {
+    private HttpResponse<InputStream> callRemoteEndpointWithRetry(HttpClient client, String decodedUrl) throws Exception {
+        return resilientOperationExecutor.execute((CheckedSupplier<HttpResponse<InputStream>>) () -> {
+            var response = client.send(buildFetchFileRequest(decodedUrl), BodyHandlers.ofInputStream());
+            if (response.statusCode() / 100 != 2) {
+                try (InputStream is = response.body()) {
+                    byte[] buffer = new byte[ERROR_RESPONSE_BODY_MAX_LENGTH];
+                    int read = IOUtils.read(is, buffer);
+                    String error = new String(Arrays.copyOf(buffer, read));
+                    throw new SLException(MessageFormat.format(Messages.ERROR_FROM_REMOTE_MTAR_ENDPOINT, response.statusCode(), error));
+                }
+            }
+            return response;
+        });
+    }
+
+    protected HttpClient buildHttpClient(String decodedUrl) {
         return HttpClient.newBuilder()
-                         //ssl and authentication configuration can be done here
-                         .version(HttpClient.Version.HTTP_1_1)
+                         .version(HttpClient.Version.HTTP_2)
```

Comment:
Do all urls work with http 2?

Reply:
No, but the client will fallback to HTTP/1.1 if 2 doesn't work.